app/services/cancel_interview.rb

```diff
@@ -17,17 +19,35 @@ def save!
     auth.assert_can_set_up_interviews!(application_choice: application_choice,
                                        course_option: application_choice.current_course_option)
 
-    if ApplicationStateChange.new(application_choice).can_cancel_interview?
-      ActiveRecord::Base.transaction do
-        interview.update!(cancellation_reason: cancellation_reason,
-                          cancelled_at: Time.zone.now)
+    raise_error_if_state_transition_not_allowed!
 
-        ApplicationStateChange.new(application_choice).cancel_interview! if @application_choice.interviews.kept.none?
-      end
+    interview.cancellation_reason = cancellation_reason
+    interview.cancelled_at = Time.zone.now
+
+    if interview_validations.valid?
+      audit(@auth.actor) do
+        ActiveRecord::Base.transaction do
+          interview.save!
 
-      CandidateMailer.interview_cancelled(application_choice, interview, cancellation_reason).deliver_later
+          ApplicationStateChange.new(application_choice).cancel_interview! if @application_choice.interviews.kept.none?
+        end
+
+        CandidateMailer.interview_cancelled(application_choice, interview, cancellation_reason).deliver_later
+      end
     else
+      raise ValidationException, interview_validations.errors.map(&:message)
+    end
+  end
+
+  def raise_error_if_state_transition_not_allowed!
+    unless ApplicationStateChange.new(application_choice).can_cancel_interview?
       raise "Interview cannot be cancelled when the application_choice is in #{application_choice.status} state"
```

Comment:
```suggestion
      raise  Workflow::NoTransitionAllowed, I18n.t('activerecord.errors.models.application_choice.attributes.status.invalid_transition')
```

https://trello.com/c/raRutZDj/840-improve-statetransitionerror-messaging-in-the-api